tagless-final/src/main/java/com/github/jorgecastillo/kotlinandroid/presentation/SuperHeroesPresentation.kt

```diff
@@ -63,11 +63,13 @@ inline fun <reified F> getSuperHeroDetails(heroId: String,
     C: MonadControl<F, GetHeroDetailsContext, CharacterError> = monadControl()): HK<F, Unit> =
     C.binding {
       val ctx = C.ask().bind()
-      val result = C.handleError(getHeroDetailsUseCase(heroId), { displayErrors(ctx, it); CharacterDto() }).bind()
-      ctx.view.drawHero(SuperHeroViewModel(
-          result.id,
-          result.name,
-          result.thumbnail.getImageUrl(MarvelImage.Size.PORTRAIT_UNCANNY),
-          result.description))
+      val result = C.handleError(getHeroDetailsUseCase(heroId), { displayErrors(ctx, it); emptyList() }).bind()
+      ctx.view.drawHero(result.map {
```

Comment:
I recall this being implemented differently in one of the other modules wrapped around the monad context to make effect capturing more explicit. Since you are not surrounding this with `catch` and you are not inside a `bindingE` thrown exceptions won't be captured and raised via `MonadError`